### PR TITLE
Fixes to effect sizes t-test

### PR DESCRIPTION
**Pull request by JohnnyDoorn**
Hedges g was wrong, added help file stuff, unit test for CI effect size, improved the function to compute the CI (MBESS instead of effsize).
Added footnote to indicate what group is used to standardize for Glass delta.

### CODE REVIEW

File: JASP-Engine/JASP/R/ttestindependentsamples.R [hunk 1]
```diff
@@ -321,80 +321,59 @@ TTestIndependentSamples <- function(dataset = NULL, options, perform = "run",
...
+								}	else if (options$effectSizesType == "glassD") {
 									d <- .clean(as.numeric((ms[1] - ms[2]) / sds[2]))
 									# Should give feedback on which data is considered 2.
-								} else if (options$effectSizesCheckbox == "hedgesG") {
-									hedges_g <- .clean(as.numeric((ms[1] - ms[2]) / sdPooled)) # same as Cohen's d so far
-									# correction_J <- function(a) { gamma(a / 2) / (sqrt(a/2) * (a - 1) / 2) } # supposedly; use approx.
-									d <- 1 - 3 / (4 * (ns[1] + ns[2]) - 9) * hedges_g # less biased / corrected version
+								} else if (options$effectSizesType == "hedgesG") {
+									a <- sum(ns) - 2
+									correction_J <- gamma(a / 2) / (sqrt(a / 2) * gamma((a - 1) / 2)) # supposedly; use approx.
```
Comment (AlexanderLyNL) [inline JASP-Engine/JASP/R/ttestindependentsamples.R]: Please use lgamma here and remove the remark regarding the approximation

File: JASP-Engine/JASP/R/ttestindependentsamples.R [hunk 2]
```diff
@@ -321,80 +321,59 @@ TTestIndependentSamples <- function(dataset = NULL, options, perform = "run",
...
-							  confIntEffSize = sort(c(ncp1*sqrt(1/ns[1]+1/ns[2]),  ncp2*sqrt(1/ns[1]+1/ns[2]) ))[order(c(1-ciEffSize, ciEffSize ))]
-							  
-							  if (direction == "greater") {
-							    confIntEffSize[2] <- Inf
-							  } else if (direction == "less") 
-							    confIntEffSize[1] <- -Inf
-							  
-							  confIntEffSize <- sort(confIntEffSize)
+								# From MBESS package by Ken Kelley, v4.6
+								dfEffSize <-  ifelse(effSize == "glass", ns[2]-1, df)
```
Comment (AlexanderLyNL) [inline JASP-Engine/JASP/R/ttestindependentsamples.R]: This works well here, but be careful with ifelse in general, as it leads to weird behaviour.

File: JASP-Engine/JASP/R/ttestindependentsamples.R [hunk 3]
```diff
@@ -778,80 +765,236 @@ TTestIndependentSamples <- function(dataset = NULL, options, perform = "run",
...
+	}
+
+	# General stop checks.
+	if(df <= 0) stop("The degrees of freedom must be some positive value.", call.=FALSE)
+
+	if(abs(ncp) > 37.62) print("The observed noncentrality parameter of the noncentral t-distribution has exceeded 37.62 in magnitude (R's limitation for accurate probabilities from the noncentral t-distribution) in the function's iterative search for the appropriate value(s). The results may be fine, but they might be inaccurate; use caution.")
+
+	if(sup.int.warns==TRUE) Orig.warn <- options()$warn; options(warn=-1)
+
+	if(!is.null(conf.level) & is.null(alpha.lower) & !is.null(alpha.upper)) stop("You must choose either to use \'conf.level\' or define the \'lower.alpha\' and \'upper.alpha\' values; here, \'upper.alpha\' is specified but \'lower.alpha\' is not", call.=FALSE)
```
Comment (AlexanderLyNL) [inline JASP-Engine/JASP/R/ttestindependentsamples.R]: The message confused me. Perhaps the following could do:

stop("You must either use \'conf.level\', or define the \'lower.alpha\' and \'upper.alpha\' values; here, \'upper.alpha\' is specified but \'lower.alpha\' is not", call.=FALSE)

File: JASP-Engine/JASP/R/ttestindependentsamples.R [hunk 4]
```diff
@@ -778,80 +765,236 @@ TTestIndependentSamples <- function(dataset = NULL, options, perform = "run",
...
+	if(!is.null(conf.level) & !is.null(alpha.lower) & is.null(alpha.upper)) stop("You must choose either to use \'conf.level\' or define the \'lower.alpha\' and \'upper.alpha\' values; here, \'lower.alpha\' is specified but \'upper.alpha\' is not", call.=FALSE)
```
Comment (AlexanderLyNL) [inline JASP-Engine/JASP/R/ttestindependentsamples.R]: stop("You must either use \'conf.level\', or define the \'lower.alpha\' and \'upper.alpha\' values; here, \'lower.alpha\' is specified but \'upper.alpha\' is not", call.=FALSE)

File: Resources/Help/analyses/ttestindependentsamples.md [hunk 5]
```diff
@@ -33,7 +33,10 @@ The independent samples t-test allows the user to estimate the effect size and t
...
   - Confidence interval: Confidence interval for the location parameter. By default, the confidence interval is set to 95%. This can be changed into the desired percentage.
-- Effect Size: For the Student t-test and Welch t-test, the effect size is given by Cohen's d; for the Mann-Whitney test, the effect size is given by the rank biserial correlation. 
+- Effect Size: For the Student t-test and Welch t-test, the effect size can be selected below; for the Mann-Whitney test, the effect size is given by the rank biserial correlation. 
+  - Cohen's d: Uses the pooled standard deviation to standardize the mean difference.
+  - Glass' delta: Uses the standard deviation of group 2 to standardize the mean difference. In order to change which group is used as group 2, you can change the order of the levels by clicking on the name of the grouping variable in the data window, click on one of the levels and then click the arrow buttons to switch the order.
+  - Hedges' g: Applies a correction factor to Cohen's d to make it less biased.
```
Comment (AlexanderLyNL) [inline Resources/Help/analyses/ttestindependentsamples.md]: Not less but just unbiased. 

- Hedges' g: Applies a correction factor to Cohen's d to make it unbiased.

File: JASP-Engine/JASP/R/ttestindependentsamples.R (hunk 4 above)
Comment (JohnnyDoorn) [reply JASP-Engine/JASP/R/ttestindependentsamples.R]: I just copied the function from MBESS to stay more true to it, but now I removed some redundant stuff, such as these things.

Comment (AlexanderLyNL) [reply JASP-Engine/JASP/R/ttestindependentsamples.R]: No worries, I was a bit confused. Just keep it as is. I'll have to think about this on a higher level; dependencies and all.

File: Resources/Help/analyses/ttestindependentsamples.md (hunk 5 above)
Comment (AlexanderLyNL) [reply Resources/Help/analyses/ttestindependentsamples.md]: This should be "unbiased" not less biased.

Comment (JohnnyDoorn) [reply Resources/Help/analyses/ttestindependentsamples.md]: This is fixed now.

**Comment by boutinb** [comment]
@JohnnyDoorn: moet dat in de 0.10.1?

**Comment by TimKDJ** [comment]
So we're good to merge now @JohnnyDoorn @AlexanderLyNL ?

**Comment by JohnnyDoorn** [comment]
Yes!

**Pull request merged**